dashboard/postgres_replicator/logger.go

```diff
@@ -0,0 +1,26 @@
+package transfer
+
+import (
+	"log"
+	"os"
+)
+
+// Logger proves per-table logging.
+type Logger struct {
+	prefix string
+	*log.Logger
+}
+
+// NewLogger returns a new Logger.
+func NewLogger(tableName string) *Logger {
+	prefix := "[" + tableName + "]"
```

Comment:
I don't think there is a reason to save the prefix in the logger, and to change it during operation. You can always print with format + "[ERROR] " to produce the error string you are trying to produce here.

The prefix you have now is also illogical, since you always add " " to it. You should have fmt.Sprintf("[%s] ", tableName) and use that as the prefix.

Reply:
The reason for this weird logic was this.
1. Normal logs would have the prefix `[tableName]_` (underscore where the space would be)
2. For error messages, I would put the error tag after the table name, like this: `[tableName][ERROR]_`, and so the location of the space would change to after the error tag.
3. Then the normal tag would need to be restored: `[tableName]_`

The space after the tags had to be moved around. But it's simpler to add the error tag before the table name so I just changed it to do that instead. It removes all that weird space stuff. Also the Logger no longer stores the prefix; I didn't realize there was a built in function.